thijsvanloef/palworld-server-docker
# Add env to toggle to automatically generate palworldsettings

## Context 

resolves #326 

## Choices

Add an env var which is false by default that decides if the PalWorldSettings.ini is compiled from env vars automatically

## Test instructions

#### Test Case: New Server, no automatic settings
1. Set `DISABLE_GENERATE_SETTINGS` to `true`
2. Start the server
3. On the server run `cat /palworld/Pal/Saved/Config/LinuxServer/PalWorldSettings.ini`
4. Verify that all values are default

#### Test Case: Existing Server, no automatic settings
1. Set `DISABLE_GENERATE_SETTINGS` to `true`
2. Modify or set `SERVER_NAME` environment variable
3. Start the server
4. On the server run `cat /palworld/Pal/Saved/Config/LinuxServer/PalWorldSettings.ini`
5. Verify that all values are still the same to prior launch and `SERVER_NAME` was not applied

#### Test Case: New Server, automatic settings
1. Set `DISABLE_GENERATE_SETTINGS` to `false` (or remove it entirely)
2. Start the server
3. On the server run `cat /palworld/Pal/Saved/Config/LinuxServer/PalWorldSettings.ini`
4. Verify that all values are as they were set as environment variables 

#### Test Case: Existing Server, automatic settings
1. Set `DISABLE_GENERATE_SETTINGS` to `false` (or remove it entirely)
2. Start the server
3. On the server run `cat /palworld/Pal/Saved/Config/LinuxServer/PalWorldSettings.ini`
4. Verify that all values are as they were set as environment variables 

## Checklist before requesting a review
